pandas/core/internals/managers.py

```diff
@@ -1525,7 +1526,9 @@ def _insert_update_mgr_locs(self, loc) -> None:
         When inserting a new Block at location 'loc', we increment
         all of the mgr_locs of blocks above that by one.
         """
-        for blkno, count in _fast_count_smallints(self.blknos[loc:]):
+        # Faster version of set(arr) for sequences of small numbers
+        blknos = np.bincount(self.blknos[loc:]).nonzero()[0]
```

Comment:
What is the advantage of this over np.unique?

Reply:
Its a nice idea but unless this is getting called in a loop I'd have a slight preference to just go with `np.unique` since its more expressive. Seems like we might be relying on some implementation details with bincount that could be hard to generalize

Though if you really want this I'd also say its not a blocker for me

Reply:
Since this was just transferring over the logic from `_fast_count_smallints` I would prefer keeping this as-is in this PR but not opposed to changing this to `np.unique` in the future